src/python/gudhi/tensorflow/ripsnet.py

```diff
@@ -0,0 +1,209 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):    Felix Hensel, Mathieu Carrière
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import tensorflow as tf
+
+class DenseRagged(tf.keras.layers.Layer):
+    """
+    This is a class for the ragged layer in the RipsNet architecture, processing the input pointclouds.
+    """
+
+    def __init__(self, units, input_dim=None, use_bias=True, activation='gelu', kernel_initializer=None, bias_initializer=None, **kwargs):
+        """
+        Constructor for the DenseRagged class.
+
+        Parameters:
+            units (int): number of units in the layer.
+            use_bias (bool): flag, indicating whether to use bias or not.
+            activation (string or function): identifier of a keras activation function, e.g. 'relu'.
+            kernel_initializer: tensorflow kernel initializer.
+            bias_initializer: tensorflow bias initializer.
+        """
+        super().__init__(dynamic=True, **kwargs)
+        self._supports_ragged_inputs = True
+        self.units = units
+        self.use_bias = use_bias
+        self.activation = tf.keras.activations.get(activation)
+        self.kernel_initializer = kernel_initializer
+        self.bias_initializer = bias_initializer
+
+    def build(self, input_shape):
+        last_dim = input_shape[-1]
+        self.kernel = self.add_weight('kernel', shape=[last_dim, self.units], trainable=True, initializer=self.kernel_initializer)
+        if self.use_bias:
+            self.bias = self.add_weight('bias', shape=[self.units, ], trainable=True, initializer=self.bias_initializer)
+        else:
+            self.bias = None
+        super().build(input_shape)
+
+    def call(self, inputs):
+        """
+        Apply DenseRagged layer on a ragged input tensor.
+
+        Parameters:
+            ragged tensor (e.g. containing a point cloud).
+
+        Returns:
+            ragged tensor containing the output of the layer.
+        """
+        outputs = tf.ragged.map_flat_values(tf.matmul, inputs, self.kernel)
+        if self.use_bias:
+            outputs = tf.ragged.map_flat_values(tf.nn.bias_add, outputs, self.bias)
+        outputs = tf.ragged.map_flat_values(self.activation, outputs)
+        return outputs
+
+
+class DenseRaggedBlock(tf.keras.layers.Layer):
```

Comment:
Indeed, I have added changed `TFBlock` so that it supports ragged inputs if the first layer is an instance of `DenseRagged`. So, indeed I think that `DenseRaggedBlock` is no longer needed. I have commented it for now but if the change is confirmed it can be deleted.